overlord/hookstate/ctlcmd/model.go

```diff
@@ -101,74 +99,120 @@ func (c *modelCommand) reportError(format string, a ...interface{}) {
 	w.Flush()
 }
 
-// checkGadgetOrModel verifies that the current snap context is either a gadget snap or that
-// the snap shares the same publisher as the current model assertion.
-func (c *modelCommand) checkGadgetOrModel(st *state.State, snapName string) error {
-	st.Lock()
-	defer st.Unlock()
+func interfaceConnected(st *state.State, snapName, ifName string) bool {
+	conns, err := ifacerepo.Get(st).Connected(snapName, ifName)
```

Comment:
we want to follow the similar code in refresh and mount and use ifacestate.ConnectionStates

Reply:
Ah right, thanks for pointing this out - I have redone the code to match what we do in refresh